dealiiadapter.cc

```diff
@@ -770,6 +791,124 @@ void ElasticProblem<dim>::output_results(const unsigned int timestep) const
     std::cout<< "\t Output written to solution-" + std::to_string(timestep) + ".vtk \n" <<std::endl;
 }
 
+template <int dim>
+void ElasticProblem<dim>::configure_precice()
+{
+    precice.configure(parameters.config_file);
+    meshID = precice.getMeshID(parameters.mesh);
+    read_data_ID  = precice.getDataID(parameters.read_data_name, meshID);
+    write_data_ID = precice.getDataID(parameters.write_data_name, meshID);
+}
+
+template <int dim>
+void ElasticProblem<dim>::initialize_precice()
+{
+    //get n_interface_nodes
+    std::set<types::boundary_id> couplingBoundary;
+    couplingBoundary.insert(parameters.interface_mesh_ID);
+    const FEValuesExtractors::Scalar x_displacement(0);
+
+    DoFTools::extract_boundary_dofs(dof_handler, fe.component_mask(x_displacement), coupling_dofs, couplingBoundary);
+    n_interface_nodes = coupling_dofs.n_elements();//TODO: Adapt
+
+    write_data = new double [dim * n_interface_nodes];
+    read_data  = new double [dim * n_interface_nodes];
+    interface_nodes_positions = new double [dim*n_interface_nodes];
+    interface_nodes_IDs       = new int [n_interface_nodes];
+
+    // get nodes_positions
+    std::map<types::global_dof_index, Point<dim>> support_points;
+    DoFTools::map_dofs_to_support_points(MappingQ1<dim>(), dof_handler, support_points);
+    int position_iterator=0;
+    for (IndexSet::ElementIterator element=coupling_dofs.begin(); element!=coupling_dofs.end(); element++)
```

Comment:
Please default to the prefix increment `++element`  (especially in loops)

The suffix versions can be costly depending on the type, the implementation, the C++ standard, the compiler and the compiler settings. It just saves you from trouble if you default to the prefix version.